tests/test_simulate.py

```diff
@@ -0,0 +1,118 @@
+import os
+import shutil
+
+import numpy as np
+import pytest
+
+import pyfstat
+
+
+@pytest.fixture
+def data_params():
+    return {
+        "fMin": 10,
+        "Band": 1,
+        "sqrtSX": dict.fromkeys(["H1", "L1"], 1e-23),
+        "timestamps": dict.fromkeys(["H1", "L1"], 1000000000 + 1800 * np.arange(5)),
+        "Tsft": 1800,
+        "randSeed": 314192,
+    }
+
+
+# @pytest.fixture
+# def signal_parameters(obsrun_parameters):
+#    return {
+#        "F0": mfd_parameters["fMin"] + 0.5 * mfd_parameters["Band"],
+#        "F1": 0.0,
+#        "Alpha": 0,
+#        "Delta": 0,
+#        "refTime": mfd_parameters["timestamps"][0],
+#        "h0": 1e-23,
+#        "cosi": 1.0,
+#        "psi": 0,
+#        "phi0": 0,
+#    }
+#
+# @pytest.fixture
+# def noise_writer(obsrun_parameters, tmp_path):
+#    writer_parameters = {**obsrun_parameters}
+#
+#    for new, old in [("tref", "refTime")]:
+#        writer_parameters[new] = w_parameters.pop(old)
+#    writer_parameters["detectors"] = ",".join(writer_parameters.pop("detectors"))
+#
+#    this_writer = pyfstat.Writer(**writer_parameters, outdir=tmp_path, label="NoiseOnly")
+#    return this_writer
+#
+#
+# @pytest.fixture
+# def signal_writer(obsrun_parameters, signal_parameters, tmp_path):
+#    writer_parameters = {**obsrun_parameters, **signal_parameters}
+#
+#    for remove in ["fMin"]:
+#        writer_parameters.pop(remove)
+#    for new, old in [("tref", "refTime"), ("phi", "phi0")]:
+#        writer_parameters[new] = w_parameters.pop(old)
+#    writer_parameters["detectors"] = ",".join(writer_parameters.pop("detectors"))
+#
+#    this_writer = pyfstat.Writer(**writer_parameters, outdir=tmp_path, label="NoiseAndSignal")
+#    return this_writer
+
+
+def test_MakeFakeData_set_data_params(data_params):
+
+    mfd = pyfstat.MakeFakeData()
+    mfd.set_data_params(**data_params)
+    for key in ["randSeed", "fMin", "Band"]:
+        assert getattr(mfd.data_params, key) == data_params[key]
+
+    detectors = list(data_params["timestamps"].keys())
+    num_detectors = len(detectors)
+
+    assert mfd.data_params.multiTimestamps.length == num_detectors
+    
+    for ifo_ind, ifo in enumerate(detectors):
+        # This fails for some reason...?
+        print(mfd.data_params.multiTimestamps.data[0].data[0])
```

Comment:
But this, as opposed to [the previous case](https://github.com/PyFstat/PyFstat/pull/382/files#r963821345), doesn't.

@dbkeitel